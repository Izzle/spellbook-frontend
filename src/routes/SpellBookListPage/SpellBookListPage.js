import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import SpellBookLink from '../../components/SpellBookLink/SpellBookLink';
import SpellApiService from '../../services/spell-api-service';
import SpellBookListContext from '../../contexts/SpellBookListContext';
import './SpellBookListPage.css';

export default class SpellBookListPage extends Component {

  static contextType = SpellBookListContext;
  
  componentDidMount() {
    // service, context, error
    SpellApiService.getAllSpellBooks()
      .then(this.context.setSpellBookList)
      .catch(this.context.setError);
  }
  
  render() {
    const { spellbookList = [] } = this.context;

    return (
      <section className='SpellBookListPage'>
        <div className='SpellBookListPage__container'>
          <h2>View your Spellbook selections here!</h2>
          {spellbookList.map((spellbook, idx )=> {
            return <SpellBookLink
                      key={idx}
                      id={spellbook.id}
                      name={spellbook.spellbook_name}/>
          })}
        </div>
      </section>
    )
  }
}