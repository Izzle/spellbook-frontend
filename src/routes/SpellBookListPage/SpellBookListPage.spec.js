import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import SpellBookListPage from './SpellBookListPage';
import { MemoryRouter } from 'react-router-dom';


describe('<SpellBookListPage />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter><SpellBookListPage /></MemoryRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<MemoryRouter><SpellBookListPage /></MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});