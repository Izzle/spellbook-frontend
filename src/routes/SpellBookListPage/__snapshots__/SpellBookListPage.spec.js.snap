// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpellBookListPage /> renders this UI as expected 1`] = `
<section
  className="SpellBookListPage"
>
  <div
    className="SpellBookListPage__container"
  >
    <h2>
      Select a Spellbook
    </h2>
  </div>
</section>
`;
